### PR TITLE
1561 exception when creating a neighborhood

### LINKED ISSUE #1561: Exception when creating a neighborhood
Project from this issue: https://github.com/Open-Systems-Pharmacology/MoBi/issues/1497

In the "Diagram" view, start creating a neighborhood from "containerB" to "TopContainer":

Application:
MoBi® 12.0.340

Object reference not set to an instance of an object.

Stack trace:
```
at OSPSuite.Core.Domain.ObjectPathFactory.addPathEntry(IEntity entity, ObjectPath objectPath)
   at OSPSuite.Core.Domain.ObjectPathFactory.CreateAbsoluteObjectPath(IEntity entity)
   at MoBi.Presentation.Tasks.Interaction.InteractionTasksForNeighborhood.Add(IContainer firstNeighbor, IContainer secondNeighbor)
   at MoBi.Presentation.Presenter.SpaceDiagram.SpatialStructureDiagramPresenter.Link(IBaseNode node1, IBaseNode node2, Object portObject1, Object portObject2)
   at MoBi.UI.Views.BaseDiagram.MoBiBaseDiagramView.onLinkCreated(GoSelectionEventArgs e)
   at MoBi.UI.Views.BaseDiagram.MoBiBaseDiagramView.<>c__DisplayClass5_0.<InitializeResources>b__4()
   at OSPSuite.Utility.Extensions.ExceptionExtensions.DoWithinExceptionHandler(Object callerObject, Action actionToExecute)
```

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #1561

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please mark relevant options with an `x` in the brackets.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: src/MoBi.Presentation/Presenter/EditFormulaInPathAndValuesPresenter.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public interface IEditFormulaInPathAndValuesPresenter : ICommandCollectorPresent
    {
       void Init<TBuildingBlock, TBuilder>(TBuilder formulaOwner, TBuildingBlock buildingBlock, UsingFormulaDecoder formulaDecoder) 
          where TBuilder : PathAndValueEntity, IUsingFormula, IWithDisplayUnit
-         where TBuildingBlock : PathAndValueEntityBuildingBlock<TBuilder>;
+         where TBuildingBlock : class, ILookupBuildingBlock<TBuilder>;
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Presenter/EditFormulaInPathAndValuesPresenter.cs:22]: This is unrelated, but I was suddenly not able to edit InitialConditions building blocks after yesterday.

**Comment by rwmcintosh** [comment]
We need a way to go back and forth between paths and container nodes in a diagram when there is no entity in the project that matches. This is because we are now drawing `ParentPath` containers in the diagram when there is no container present for that path.

File: src/MoBi.UI/Diagram/DiagramManagers/SpatialStructureDiagramManager.cs [hunk 2]
```diff
@@ -95,5 +96,33 @@ public override IDiagramManager<MoBiSpatialStructure> Create()
...
+      protected override (string, string) PathsForNeighborhood(INeighborhoodBase neighborhoodBase)
+      {
+         if (!(neighborhoodBase is NeighborhoodBuilder builder))
+         {
+            return base.PathsForNeighborhood(neighborhoodBase);
+
+         }
+
+         return (builder.FirstNeighbor == null ? builder.FirstNeighborPath : builder.FirstNeighbor.EntityPath(),
+            builder.SecondNeighbor == null ? builder.FirstNeighborPath : builder.SecondNeighbor.EntityPath());
```
Comment (Yuri05) [inline src/MoBi.UI/Diagram/DiagramManagers/SpatialStructureDiagramManager.cs:109]: I assume it should be `? builder.SecondNeighborPath` in the last line

File: src/MoBi.UI/Diagram/DiagramManagers/SpatialStructureDiagramManager.cs [hunk 3]
```diff
@@ -95,5 +96,33 @@ public override IDiagramManager<MoBiSpatialStructure> Create()
       {
          return new SpatialStructureDiagramManager();
       }
+
+      protected override (string, string) PathsForNeighborhood(INeighborhoodBase neighborhoodBase)
+      {
+         if (!(neighborhoodBase is NeighborhoodBuilder builder))
+         {
+            return base.PathsForNeighborhood(neighborhoodBase);
+
```
Comment (Yuri05) [inline src/MoBi.UI/Diagram/DiagramManagers/SpatialStructureDiagramManager.cs:105]: remove empty line

**Review by Yuri05 — APPROVED** [review]
Can be merged after the comments are addressed

**Pull request merged**